screens/GroupInfo.js

```diff
@@ -60,31 +61,54 @@ class GroupInfo extends Component {
       var usersArray = [];
       const groupID = this.state.groupID;
       this.setState({usernames: []})
-      const userId = firebase.auth().currentUser.uid;
+      //TODO remove after dev
+      const userId = firebase.auth().currentUser ? firebase.auth().currentUser.uid : "testAdminId"
       this.setState({userID: userId})
 
       firebase.database().ref('/').on('value', (snapshot) => {
         const firebaseDB = snapshot.toJSON();
         usersArray = []
-        for (var user in firebaseDB.groups[groupID].groupMemberIds){
-            if (firebaseDB.users.hasOwnProperty(user)) {   
-                usersArray.push(firebaseDB.users[user].first_name + ' ' + firebaseDB.users[user].last_name);
-            }
-        }
-        const moment = require('moment');
-        let today = moment().format('YYYY/MM/DD');
-        today = today.split('/').join('');
-        this.setState({usernames: usersArray});
-        this.setState({group: firebaseDB.groups[groupID]});
-
-        if(firebaseDB.groups[groupID].groupMemberIds[userId].hasOwnProperty(today)){
-            this.setState({complete: firebaseDB.groups[groupID].groupMemberIds[userId][today]})
-        }
 
+        //makes sure group hasn't been deleted 
+          if (firebaseDB.groups.hasOwnProperty(groupID) && firebaseDB.groups[groupID].groupMemberIds.hasOwnProperty(userId)) {
+              for (var user in firebaseDB.groups[groupID].groupMemberIds) {
+                  if (firebaseDB.users.hasOwnProperty(user)) {
+                      usersArray.push(firebaseDB.users[user].first_name + ' ' + firebaseDB.users[user].last_name);
+                  }
+              }
+              const moment = require('moment');
+              let today = moment().format('YYYY/MM/DD');
+              today = today.split('/').join('');
+              this.setState({ usernames: usersArray });
+              this.setState({ group: firebaseDB.groups[groupID] });
+             
+              if (firebaseDB.groups[groupID].groupMemberIds[userId].hasOwnProperty(today)) {
+                  this.setState({ complete: firebaseDB.groups[groupID].groupMemberIds[userId][today] })
+              }
+          }
       });
-      
-      
     }
+
+    gotTodashboard = () => {this.props.navigation.navigate('Dashboard')}
+
+    setModalVisible = (isVis) => {
+        this.setState({mvis: isVis})
+    }
+    
+    leaveGroup = () => {
+        const groupID = this.state.groupID;
+        if (this.state.usernames.length == 1) {
+            const dbGroup = firebase.database().ref('/groups/' + groupID);
+            dbGroup.remove();
+        } else {
+            //TODO remove after dev
+            const userId = firebase.auth().currentUser ? firebase.auth().currentUser.uid : "testAdminId"
```

Comment:
Would it make more sense to store userId as a state variable so that we don't repeatedly call to check?

Reply:
Yeah that's what I had originally, but I was getting some weird warnings so i switched it back.